Sources/MacOSPlatform/MacOS.swift

```diff
@@ -0,0 +1,184 @@
+import Foundation
+import SwiftlyCore
+
+struct SwiftPkgInfo: Codable {
+    var CFBundleIdentifier: String
+}
+
+/// `Platform` implementation for macOS systems.
+public struct MacOS: Platform {
+    public init() {}
+
+    public var appDataDirectory: URL {
+        return FileManager.default.homeDirectoryForCurrentUser
+            .appendingPathComponent("Library/Application Support", isDirectory: true)
+    }
+
+    public var toolchainFileExtension: String {
+        "pkg"
+    }
+
+    public func isSystemDependencyPresent(_: SystemDependency) -> Bool {
+        // All system dependencies on macOS should be present
+        true
+    }
+
+    public func verifySystemPrerequisitesForInstall(requireSignatureValidation: Bool) throws {
+        // All system prerequisites should be there for macOS
+    }
+
+    public func install(from tmpFile: URL, version: ToolchainVersion) throws {
+        guard tmpFile.fileExists() else {
+            throw Error(message: "\(tmpFile) doesn't exist")
+        }
+
+        if !self.swiftlyToolchainsDir.fileExists() {
+            try FileManager.default.createDirectory(at: self.swiftlyToolchainsDir, withIntermediateDirectories: false)
```

Comment:
Is this necessary given the installer will be creating this folder anyway

Reply:
The installer won't be running in certain cases, such as when the swiftly binary is being tested in dev, or if the swiftly binary is distributed through other means, such as packages. On macOS the testing is done manually at the moment. With swirly self-install and the init subcommand this gets centralized more.